# Задачи по Эконометрике: Интерпретация коэффициентов
# Wage Equation

## Wage Equation (Модель 1)

На датасете `wage2` рассмотрим регрессию **log(wage) на age, IQ**

Спецификация: $log(wage)=\beta_0+\beta_1age+\beta_2IQ+u$

In [2]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [4]:
df = pd.read_csv('/Users/ulianatararikina/Downloads/Econometrica-master-2/python-notebooks/data-csv/wage2.csv')
df.head()

wage  hours   IQ  KWW  educ  exper  tenure  age  married  black  south  \
0   769     40   93   35    12     11       2   31        1      0      0   
1   808     50  119   41    18     11      16   37        1      0      0   
2   825     40  108   46    14     11       9   33        1      0      0   
3   650     40   96   32    12     13       7   32        1      0      0   
4   562     40   74   27    11     14       5   34        1      0      0   

   urban  sibs  brthord  meduc  feduc     lwage  
0      1     1      2.0    8.0    8.0  6.645091  
1      1     1      NaN   14.0   14.0  6.694562  
2      1     1      2.0   14.0   14.0  6.715384  
3      1     4      3.0   12.0   12.0  6.476973  
4      1    10      6.0    6.0   11.0  6.331502

### Cпособ оценивания через спецификацию

In [7]:
# специфицируем модель через формулу
wage_eq1 = smf.ols(formula='np.log(wage)~age+IQ', data=df).fit()
# Коэфициенты модели с округление
wage_eq1.params.round(3)

Intercept    5.077
age          0.024
IQ           0.009
dtype: float64

In [11]:
# Альтеративно
wage_eq2 = sm.OLS.from_formula(formula='lwage~age+IQ', data=df).fit()
# Коэфициенты модели с округление
wage_eq2.params.round(3)

Intercept    5.077
age          0.024
IQ           0.009
dtype: float64

### Интерпретация модели
1) Увеличение результата теста IQ на 1 балл прирощяет месячную зарплату на 0.9%
2) Каждый год месячная зарплата человека в среднем увеличивается на 2.4%

## Wage Equation (Модель 2)

На датасете `wage2` рассмотрим региессию **log(wage) на age, IQ, south, urban, married, KWW**

Спецификация: $log(wage)=\beta_0+\beta_1age+\beta_2IQ+\cdots+u$

In [14]:
# Описательные статистики по переменным модели
df[['lwage', 'age', 'IQ', 'south', 'urban', 'married', 'KWW']].describe()

lwage         age          IQ       south       urban     married  \
count  935.000000  935.000000  935.000000  935.000000  935.000000  935.000000   
mean     6.779004   33.080214  101.282353    0.341176    0.717647    0.893048   
std      0.421144    3.107803   15.052636    0.474358    0.450385    0.309217   
min      4.744932   28.000000   50.000000    0.000000    0.000000    0.000000   
25%      6.505783   30.000000   92.000000    0.000000    0.000000    1.000000   
50%      6.807935   33.000000  102.000000    0.000000    1.000000    1.000000   
75%      7.056175   36.000000  112.000000    1.000000    1.000000    1.000000   
max      8.032035   38.000000  145.000000    1.000000    1.000000    1.000000   

              KWW  
count  935.000000  
mean    35.744385  
std      7.638788  
min     12.000000  
25%     31.000000  
50%     37.000000  
75%     41.000000  
max     56.000000

### Cпособ оценивания через спецификацию

In [23]:
# специфицируем моделл через формулу
wage_eq1 = smf.ols(formula='lwage~age+IQ+south+urban+married+KWW', data=df).fit()
# Коэфициенты модели с округление
wage_eq1.params.round(3)

Intercept    5.126
age          0.014
IQ           0.007
south       -0.101
urban        0.165
married      0.191
KWW          0.007
dtype: float64

In [18]:
# Число наблюдений, под которые подгонялась модель
wage_eq1.nobs

935.0

In [26]:
# Альтеративно
sleep_eq2 = sm.OLS.from_formula(formula='lwage~age+IQ+south+urban+married+KWW', data=df).fit()
# Коэфициенты модели с округление
sleep_eq2.params.round(3)

Intercept    5.126
age          0.014
IQ           0.007
south       -0.101
urban        0.165
married      0.191
KWW          0.007
dtype: float64

### Интерпретация модели
1) Увеличение результата теста IQ на 1 балл прирощяет месячную зарплату на 0.7%
2) Каждый год месячная зарплата человека в среднем увеличивается на 1.4%